great_tables/_data_color/base.py

```diff
@@ -258,21 +266,25 @@ def data_color(
         # Replace 'None' and 'np.nan' values in `color_vals` with the `na_color=` color
         color_vals = [na_color if is_na(data_table, x) else x for x in color_vals]
 
-        # for every color value in color_vals, apply a fill to the corresponding cell
-        # by using `tab_style()`
-        for i, color_val in enumerate(color_vals):
-            if autocolor_text:
-                fgnd_color = _ideal_fgnd_color(bgnd_color=color_val)
-
-                gt_obj = gt_obj.tab_style(
-                    style=[text(color=fgnd_color), fill(color=color_val)],
-                    locations=body(columns=col, rows=[i]),
-                )
-
-            else:
-                gt_obj = gt_obj.tab_style(
-                    style=fill(color=color_val), locations=body(columns=col, rows=[i])
-                )
+        # For each row, we check if the row index is selected, and then apply a fill to the
+        # corresponding cell using `tab_style()`
+        n_rows = len(resolve_rows_i(self))
+        iter_color_vals = iter(color_vals)
+        for i in range(n_rows):
```

Comment:
Would something like...

```python
for i, color_val in zip(row_pos, color_vals):
    ...
```

work? (in this case, you wouldn't need to check `if i in row_pos`)

Reply:
@machow Clever suggestion! I believe this code will work better.